test/ignore_step_error_test.go

```diff
@@ -22,85 +22,69 @@ package test
-      taskSpec:
-        params:
-        - name: param1
-        - name: param2
-        steps:
-        - name: foo
-          image: busybox
-          script: 'exit 0'`, helpers.ObjectNameForTest(t)))
-
-	if _, err := c.V1beta1PipelineRunClient.Create(ctx, pipelineRun, metav1.CreateOptions{}); err != nil {
-		t.Fatalf("Failed to create PipelineRun `%s`: %s", pipelineRun.Name, err)
+  taskSpec:
+    results:
+      - name: result1
+      - name: result2
+    steps:
+      - name: failing-step
+        onError: continue
+        image: busybox
+        script: 'echo -n 123 | tee $(results.result1.path); exit 1; echo -n 456 | tee $(results.result2.path)'
+`, taskRunName, namespace))
+
+	if _, err := c.V1beta1TaskRunClient.Create(ctx, tr, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Failed to create TaskRun: %s", err)
 	}
 
-	t.Logf("Waiting for PipelineRun in namespace %s to fail", namespace)
-	if err := WaitForPipelineRunState(ctx, c, pipelineRun.Name, timeout, FailedWithReason(pipelinerun.ReasonInvalidTaskResultReference, pipelineRun.Name), "InvalidTaskResultReference", v1beta1Version); err != nil {
-		t.Errorf("Error waiting for PipelineRun to fail: %s", err)
+	t.Logf("Waiting for TaskRun in namespace %s to finish", namespace)
+	if err := WaitForTaskRunState(ctx, c, taskRunName, TaskRunSucceed(taskRunName), "TaskRunSucceeded", v1beta1Version); err != nil {
+		t.Errorf("Error waiting for TaskRun to finish: %s", err)
 	}
 
-	taskrunList, err := c.V1beta1TaskRunClient.List(ctx, metav1.ListOptions{LabelSelector: "tekton.dev/pipelineRun=" + pipelineRun.Name})
+	taskRun, err := c.V1beta1TaskRunClient.Get(ctx, taskRunName, metav1.GetOptions{})
 	if err != nil {
-		t.Fatalf("Error listing TaskRuns for PipelineRun %s: %s", pipelineRun.Name, err)
-	}
-
-	if len(taskrunList.Items) != 1 {
-		t.Fatalf("The pipelineRun should have exactly 1 taskRun for the first task \"task1\"")
+		t.Fatalf("Couldn't get expected TaskRun %s: %s", taskRunName, err)
 	}
 
-	taskrunItem := taskrunList.Items[0]
-	if taskrunItem.Labels["tekton.dev/pipelineTask"] != "task1" {
-		t.Fatalf("TaskRun was not found for the task \"task1\"")
+	if !taskRun.Status.GetCondition(apis.ConditionSucceeded).IsTrue() {
+		t.Fatalf("Expected TaskRun %s to have succeeded but Status is %v", taskRunName, taskRun.Status)
 	}
 
-	if len(taskrunItem.Status.TaskRunResults) != 1 {
+	if len(taskRun.Status.TaskRunResults) != 1 {
 		t.Fatalf("task1 should have produced a result before failing the step")
 	}
 
-	for _, r := range taskrunItem.Status.TaskRunResults {
-		if r.Name == "result1" && r.Value.StringVal != "123" {
-			t.Fatalf("task1 should have initialized a result \"result1\" to \"123\"")
-		}
+	expectedResults := []v1beta1.TaskRunResult{{
+		Name:  "result1",
+		Type:  "string",
+		Value: *v1beta1.NewArrayOrString("123"),
+	}}
+
+	if d := cmp.Diff(taskRun.Status.TaskRunResults, expectedResults); d != "" {
+		t.Errorf("Got unexpected results %s", diff.PrintWantGot(d))
```

Comment:
I think `PrintWantGot` assumes that `want` is the first argument to `cmp.Diff`, but here you're using `got` as the first argument, so the failure output may be confusing.